reclist/abstractions.py

```diff
@@ -126,14 +126,21 @@ def __init__(self, model: RecModel, dataset: RecDataset, y_preds: list = None):
         self._y_train = dataset.y_train
         self._x_test = dataset.x_test
         self._y_test = dataset.y_test
-        self._y_preds = y_preds if y_preds else model.predict(dataset.x_test)
+        self._y_preds = y_preds
         self.rec_model = model
         self.product_data = dataset.catalog
         self._test_results = []
         self._test_data = {}
         self._dense_repr = {}
 
-        assert len(self._y_test) == len(self._y_preds)
+        if self._y_preds is not None:
+            assert len(self._y_test) == len(self._y_preds)
+
+    def get_y_preds(self, x_test, y_test, overwrite=False):
```

Comment:
I am not super sure about this method and the overwrite behavior, seems order dependent and the users might get confused about this?

Reply:
Yeah agree that it's order dependent. I think there's probably a better way of writing this. My consideration is mainly compute cost and time - we don't want to regenerate predictions for each test in the reclist, but generate new predictions as appropriate.

Reply:
Right, but it's not clear which is the "first" prediction set and when this should be overwritten (thinking from a user perspective).

A solution might be to have different variables for different prediction sets, in this way independently from the order in which the test runs, the results is always the same. What so you think?

Reply:
We will return to this when refactoring the dataset abstraction!